File: 605-can-place-flowers/solution.py
class Solution(object):
    def canPlaceFlowers(self, flowerbed, n):
        """
        :type flowerbed: List[int]
        :type n: int
        :rtype: bool
        """
        count, i = 0, 0
            
        while i <= len(flowerbed) - 1:
            if flowerbed[i] == 0 and (i == 0 or flowerbed[i-1] == 0) and (i == len(flowerbed)-1 or flowerbed[i+1] == 0):
                flowerbed[i], count = 1, count + 1
            if count >= n:
                return True
            i += 1

        return False
    

File: 605-can-place-flowers/solution.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int count = 0, i = 0;
        while (i <= flowerbed.size() - 1) {
            if (flowerbed[i] == 0 &&
                (i == 0 || flowerbed[i-1] == 0) &&
                (i == flowerbed.size() - 1 || flowerbed[i+1] == 0)
            ) {
                flowerbed[i] = 1;
                count += 1;
            }
            i += 1;
            if (count >= n) return true;
        }
        return false;
    }
};
